### ISSUE TITLE
bool and number.Real are incomparable

### ISSUE THREAD

**Issue by coder0xff** [python/mypy#10017]
**Bug Report**

The following produces an unexpected error.

    def foo(a: bool, b: Real):
        return a < b  # Unsupported operand types for < ("bool" and "Real")

Of note, reversing the types doesn't result in an error

**To Reproduce**

Run mypy on the above

**Expected Behavior**

No mypy error

**Actual Behavior**

A mypy error is generated. `Unsupported operand types for < ("bool" and "Real")`

**Your Environment**

- Mypy version used: 0.800
- Python version used: 3.8
- Operating system and version: Ubuntu 18.04.5 LTS

**Comment by JelleZijlstra** [comment]
This is an aspect of #3186. (Though in this particular case, I feel like it *should* be an error; it's hard to imagine a legitimate use case for checking whether a bool is lesser than an arbitrary Real.)

**Comment by coder0xff** [comment]
Seems counterintuitive perhaps, but the comparison is well defined nonetheless.

**Comment by posita** [comment]
> This is an aspect of #3186. (Though in this particular case, I feel like it _should_ be an error; it's hard to imagine a legitimate use case for checking whether a bool is lesser than an arbitrary Real.)

Except that both Python and Mypy already allow this without error:

``` python
def bool_float_cmp(a: bool, b: float) -> bool:
    return b >= a
```

``` python
>>> 1.0 > True
False
>>> 1.1 > True
True
```

These, however, are even harder to dismiss:

``` python
from numbers import Integral, Rational, Real

def integral_int_cmp(a: int, b: Integral) -> bool:
    return b >= a  # Unsupported operand types for <= ("int" and "Integral")

def integral_float_cmp(a: float, b: Integral) -> bool:
    return b >= a  # Unsupported operand types for <= ("float" and "Integral")

def rational_int_cmp(a: int, b: Rational) -> bool:
    return b >= a  # Unsupported operand types for <= ("int" and "Rational")

def rational_float_cmp(a: float, b: Rational) -> bool:
    return b >= a  # Unsupported operand types for <= ("float" and "Rational")

def real_int_cmp(a: int, b: Real) -> bool:
    return b >= a  # Unsupported operand types for <= ("int" and "Real")

def real_float_cmp(a: float, b: Real) -> bool:
    return b >= a  # Unsupported operand types for <= ("float" and "Real")
```

**Comment by posita** [comment]
@coder0xff, besides `# type: ignore`, there's always this approach:

``` python
from numbers import Integral, Rational, Real
from operator import __ge__

def integral_int_cmp(a: int, b: Integral) -> bool:
    return __ge__(b, a)

def integral_float_cmp(a: float, b: Integral) -> bool:
    return __ge__(b, a)

def rational_int_cmp(a: int, b: Rational) -> bool:
    return __ge__(b, a)

def rational_float_cmp(a: float, b: Rational) -> bool:
    return __ge__(b, a)

def real_int_cmp(a: int, b: Real) -> bool:
    return __ge__(b, a)
```

At first blush, they look pretty similar, performance-wise:

``` python
In [7]: from fractions import Fraction

In [8]: from operator import __ge__

In [9]: %timeit b >= a
3.62 µs ± 8.07 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [10]: %timeit __ge__(b, a)
3.63 µs ± 6.16 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```
